src/Transit.js

```diff
@@ -1,13 +1,78 @@
 import BusStop from './BusStop.js'
+import ProgressCircle from './ProgressCircle.js';
+import { useEffect, useState } from 'react';
 
 
 function Transit() {
+
+    const busStops = ["1328", "1098", "53342"];
+    const [busStopsData, setBusStopsData] = useState({});
+
+    const fetchSingleStopDepartures = async (busStopId) => {
+      const url=`https://svc.metrotransit.org/nextripv2/${busStopId}`;
+
+      const response = await fetch(url);
+      const json = await response.json();
+      const realTimeDepartures = [];
+      const scheduledDepartures = [];
+  
+      for (let i = 0; i < json.departures.length; i++) {
+        if (json.departures[i].actual) {
+          realTimeDepartures.push(json.departures[i])
+        } else {
+          scheduledDepartures.push(json.departures[i])
+        }
+      };
   
+      const singleStopData = {
+        "realTimeDepartures": realTimeDepartures, 
+        "scheduledDepartures": scheduledDepartures
+      };
+
+      return singleStopData;
+    };
+
+    const fetchAllDepartures = async () => {
+      const tempBusStopsData = {};
+
+      for (const busStopId of busStops) {
+        tempBusStopsData[busStopId] = await fetchSingleStopDepartures(busStopId);
+      };
+
+      setBusStopsData(tempBusStopsData);
+    };
+
+    //interval for progress circle and repeated departure fetching
+    const [timeLeft, setTimeLeft] = useState(32)
+
+    useEffect(() => {
+      const interval = setInterval(() => {
+        setTimeLeft(timeLeft - 1)
+      }, 1000);
+
+      if (timeLeft === 0) {
+        fetchAllDepartures();
+        setTimeLeft(32);
+      }

```

Comment:
I think this part of the code should be in it's own use effect that's triggered on timeLeft. But, the setInterval stuff should only run the first time (i.e. a useEffect with empty list)